### DISCUSSION TITLE
error in path

### DISCUSSION THREAD

**Discussion by MohMehrnia** [PaddlePaddle/PaddleOCR#14922] [Q&A]
`Global:
  use_gpu: True  # استفاده از GPU (اگر False باشد، از CPU استفاده می‌شود)
  epoch_num: 100  # تعداد دوره‌های آموزش
  log_smooth_window: 20  # پنجره لگ‌گیری برای محاسبه میانگین نرم‌شده
  print_batch_step: 10  # هر چند batch اطلاعات چاپ شود
  save_model_dir: "./output/"  # مسیر ذخیره مدل آموزش‌دیده
  save_epoch_step: 3  # هر چند دوره مدل ذخیره شود
  eval_batch_step: [ 0, 2000 ]  # هر چند batch ارزیابی انجام شود
  cal_metric_during_train: False  # محاسبه معیارها در حین آموزش
  save_inference_dir: "./inference/"  # مسیر ذخیره مدل ONNX
  use_visualdl: False  # استفاده از VisualDL برای نمایش نتایج
  character_dict_path: "/home/mohammad/MySourceCodes/python/OCRTrain/persian_dict.txt"  # فایل دیکشنری حروف فارسی
  max_text_length: 25  # حداکثر طول متن
  character_type: "ch"  # نوع کاراکترها (فارسی)
  loss_type: "ctc"  # نوع تابع زیان (CTC)
  distort: True  # اعمال تغییرات تصادفی روی تصاویر
  image_shape: [ 3, 32, 100 ]  # اندازه تصاویر ورودی (کانال‌ها، ارتفاع، عرض)
  label_list: [ "train", "val" ]  # لیست مجموعه‌های داده (آموزش و اعتبارسنجی)
  use_parallel: False  # استفاده از آموزش موازی
PostProcess:
  name: CTCLabelDecode  # نوع پس‌پردازش برای مدل‌های مبتنی بر CTC
Loss:
  name: CTCLoss  # نوع تابع زیان برای مدل‌های مبتنی بر CTC

Optimizer:
  name: Adam  # نوع بهینه‌ساز (مانند Adam)
  lr:
    name: Cosine  # نوع جدول نرخ یادگیری
    learning_rate: 0.001  # نرخ یادگیری اولیه
    warmup_epoch: 5  # تعداد دوره‌های گرم‌کردن
    eta_min: 0.00001  # حداقل نرخ یادگیری
  regularizer:
    name: L2  # نوع منظم‌سازی
    factor: 0.00001  # ضریب منظم‌سازی
Metric:
  name: RecMetric  # نوع معیار ارزیابی برای مدل‌های تشخیص متن
  main_indicator: acc  # شاخص اصلی ارزیابی (مانند دقت)
Architecture:
  model_type: "rec"  # نوع مدل (تشخیص متن)
  algorithm: "CRNN"  # الگوریتم مورد استفاده
  transforms:
    - DecodeImage:
        img_mode: "RGB"  # حالت تصویر (RGB)
        channel_first: False  # ترتیب کانال‌ها (آخرین بعد)
    - RecResizeImg:
        image_shape: [ 3, 32, 100 ]  # اندازه تصویر ورودی
    - KeepKeys:
        keep_keys: [ "image", "label" ]  # کلیدهای مورد نیاز
  Backbone:
    name: "MobileNetV3"  # معماری بک‌بون
    scale: 0.5  # مقیاس مدل
    model_name: "large"  # نوع مدل
    disable_se: True  # غیرفعال کردن SE Block
  Neck:
    name: "SequenceEncoder"  # معماری گردن
    encoder_type: "rnn"  # نوع انکودر
    hidden_size: 96  # اندازه لایه پنهان
  Head:
    name: "CTCHead"  # معماری هد
    fc_decay: 0.00001  # ضریب منظم‌سازی در لایه‌های کاملاً متصل

Train:
  dataset:
    name: SimpleDataSet  # نوع مجموعه داده
    data_dir: "/home/mohammad/MySourceCodes/python/OCRTrain/dataset/train"  # مسیر داده‌های آموزشی
    label_file_list: [ "/home/mohammad/MySourceCodes/python/OCRTrain/dataset/train/label.txt" ]  # فایل لیست برچسب‌ها
    transforms:
      - DecodeImage:
          img_mode: "RGB"
          channel_first: False
      - RecResizeImg:
          image_shape: [ 3, 32, 100 ]
      - KeepKeys:
          keep_keys: [ "image", "label" ]
  loader:
    shuffle: True  # به هم‌ریختن داده‌ها
    drop_last: True  # حذف دسته‌های باقی‌مانده
    batch_size_per_card: 256  # اندازه دسته برای هر کارت GPU
    num_workers: 8  # تعداد worker برای بارگذاری داده‌ها

Eval:
  dataset:
    name: SimpleDataSet  # نوع مجموعه داده
    data_dir: "/home/mohammad/MySourceCodes/python/OCRTrain/dataset/val"  # مسیر داده‌های اعتبارسنجی
    label_file_list: [ "/home/mohammad/MySourceCodes/python/OCRTrain/dataset/val/label.txt" ]  # فایل لیست برچسب‌ها
    transforms:
      - DecodeImage:
          img_mode: "RGB"
          channel_first: False
      - RecResizeImg:
          image_shape: [ 3, 32, 100 ]
      - KeepKeys:
          keep_keys: [ "image", "label" ]
  loader:
    shuffle: False  # به هم‌ریختن داده‌ها
    drop_last: False  # حذف دسته‌های باقی‌مانده
    batch_size_per_card: 256  # اندازه دسته برای هر کارت GPU
    num_workers: 8  # تعداد worker برای بارگذاری داده‌ها`

i used this config.yml by show me bellow error

/home/mohammad/MySourceCodes/python/OCRTrain/dataset/train**/./dataset/train**/image_4943.jpg does not exist!

**Comment by GreatV** [comment]
The error message indicates that the training script is looking for an image at a path that appears to be incorrect due to duplication:

```
/home/mohammad/MySourceCodes/python/OCRTrain/dataset/train**/./dataset/train**/image_4943.jpg does not exist!
```

This suggests that there might be an issue with how the paths are defined in your **config.yml** file or how the dataset file (e.g., `label.txt`) specifies the image paths.

### Possible Causes and Solutions

#### 1. Verify `data_dir` Paths
✅ **Check if your `data_dir` is correctly set.**  
Your current configuration specifies:
```yaml
Train:
  dataset:
    data_dir: "/home/mohammad/MySourceCodes/python/OCRTrain/dataset/train"
```
This should be the root directory containing images. Make sure there is **no nested "dataset/train"** inside it. Also, check if the actual images exist in that directory.

✅ **Check if `label_file_list` paths are correct.**
```yaml
label_file_list: [ "/home/mohammad/MySourceCodes/python/OCRTrain/dataset/train/label.txt" ]
```
Open this file and ensure that image paths inside it are **relative** to `data_dir` OR are absolute paths.

#### 2. Check `label.txt` Image Paths
⌛ **Your `label.txt` file should contain paths like:**
```
image_4943.jpg label_text_here
image_1234.jpg label_text_here
```
Make sure there is **no unnecessary subdirectory repetition** in these paths. If your paths are absolute, ensure they are correct.

#### 3. Debug Paths
Run this snippet before training to check if the paths are resolving correctly:
```python
import os

train_data_dir = "/home/mohammad/MySourceCodes/python/OCRTrain/dataset/train"
label_file_path = "/home/mohammad/MySourceCodes/python/OCRTrain/dataset/train/label.txt"

with open(label_file_path, 'r') as f:
    lines = f.readlines()

for line in lines[:10]:  # Check a few entries
    img_path = line.split()[0]  # Assuming label.txt format is "image_name label"
    full_path = os.path.join(train_data_dir, img_path)
    if not os.path.exists(full_path):
        print(f"Missing: {full_path}")
```

If paths are incorrect, adjust them accordingly.

#### 4. Ensure No Trailing Slash Issue
🔧 **Modify paths carefully to avoid duplicate paths.**  
Try modifying `data_dir` to:
```yaml
data_dir: "/home/mohammad/MySourceCodes/python/OCRTrain/dataset"
```
And update `label.txt` paths to:
```
train/image_4943.jpg label_text_here
train/image_1234.jpg label_text_here
```
This ensures paths are resolved correctly.

---

### Additional Debugging
If the error persists, check if the dataset loader appends additional subdirectories due to incorrect configurations. Modifying or printing paths before loading may reveal inconsistencies.

Let me know if you need further clarification! 🚀

---
*Response generated by 🤖 [feifei-bot](https://github.com/GreatV/feifei) | chatgpt-4o-latest*

 ⚠️ This is an AI-generated response. Please verify any critical information.
